logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -232,6 +232,24 @@ class TestPipeline < LogStash::Pipeline
   end
 
   context "compiled flush function" do
+    describe "flusher thread" do
+      before(:each) do
+        allow(LogStash::Plugin).to receive(:lookup).with("input", "dummyinput").and_return(DummyInput)
+        allow(LogStash::Plugin).to receive(:lookup).with("codec", "plain").and_return(DummyCodec)
+        allow(LogStash::Plugin).to receive(:lookup).with("output", "dummyoutput").and_return(DummyOutput)
+      end
+
+      let(:config) { "input { dummyinput {} } output { dummyoutput {} }"}
+
+      it "should start the flusher thread only after the pipeline is running" do
+        pipeline = TestPipeline.new(config)
+
+        expect(pipeline).to receive(:transition_to_running).ordered.and_call_original
+        expect(pipeline).to receive(:start_flusher).ordered.and_call_original
```

Comment:
I think that should be a different unit test. `start_flusher` calls `flush` on a periodic basis. We don't want this test to yield a false positive in the case where somehow `flush` is only called once (or twice, or three times!) but the flusher itself is not started.

I do think we could use a separate unit test for the flusher's behavior.